### PR TITLE
support pending outcome in trx

### PR DESCRIPTION
**Pull request by tanvi-soni**

### CODE REVIEW

File: src/Agent.Worker/TestResults/TrxResultReader.cs [hunk 1]
```diff
@@ -286,14 +286,18 @@ private List<TestCaseResultData> ReadActualResults(XmlNodeList resultsNodes, str
                 DateTime completedDate = startedDate.AddTicks(duration.Ticks);
                 resultCreateModel.CompletedDate = completedDate;
 
-                if (resultNode.Attributes["outcome"] != null && resultNode.Attributes["outcome"].Value != null)
+                if (resultNode.Attributes["outcome"] == null || resultNode.Attributes["outcome"].Value == null)
+                {
+                    resultCreateModel.Outcome = "Failed";
```
Comment (vimegh) [inline src/Agent.Worker/TestResults/TrxResultReader.cs]: Modify logic and use enum TestOutcome

File: src/Test/L0/Worker/TestResults/TrxResultReaderTests.cs [hunk 2]
```diff
@@ -49,6 +49,49 @@ public void ResultsWithoutTestNamesAreSkipped()
...
+
+                 "<ResultSummary outcome=\"Failed\"><Counters total = \"2\" executed = \"2\" passed=\"1\" failed=\"1\" error=\"0\" timeout=\"0\" aborted=\"0\" inconclusive=\"0\" passedButRunAborted=\"0\" notRunnable=\"0\" notExecuted=\"0\" disconnected=\"0\" warning=\"0\" completed=\"0\" inProgress=\"0\" pending=\"0\" />" +
+                 "</ResultSummary>" +
+
+               "</TestRun>";
+
+            var runData = GetTestRunData(trxContents, null, new TestRunContext("Owner", "any cpu", "debug", 1, "", "releaseUri", "releaseEnvironmentUri"));
+          
+            Assert.Equal(runData.Results.Length, 2);
+
```
Comment (nigurr) [inline src/Test/L0/Worker/TestResults/TrxResultReaderTests.cs]: remove empty lines. approved.

**Comment by vimegh** [comment]
Approved with suggestions.

**Comment by vimegh** [comment]
sorry closed by mistake.

**Pull request merged**